Content.Client/Chemistry/Visualizers/FoamVisualizer.cs

```diff
@@ -1,68 +1,66 @@
-﻿using System;
+﻿using Content.Client.Chemistry.Components;
 using Content.Shared.Foam;
-using JetBrains.Annotations;
 using Robust.Client.Animations;
 using Robust.Client.GameObjects;
-using Robust.Shared.GameObjects;
-using Robust.Shared.IoC;
-using Robust.Shared.Maths;
-using Robust.Shared.Serialization;
-using Robust.Shared.Serialization.Manager.Attributes;
 
-namespace Content.Client.Chemistry.Visualizers
+namespace Content.Client.Chemistry.Visualizers;
+
+[RegisterComponent]
+public sealed class FoamVisualizerComponent : Component
 {
-    [UsedImplicitly]
-    public sealed class FoamVisualizer : AppearanceVisualizer, ISerializationHooks
-    {
-        private const string AnimationKey = "foamdissolve_animation";
+    public const string AnimationKey = "foamdissolve_animation";
+
+    [DataField("animationTime")]
+    [Access(typeof(FoamVisualizerSystem))]
+    public float Delay = 0.6f;
 
-        [DataField("animationTime")]
-        private float _delay = 0.6f;
+    [DataField("animationState")]
+    [Access(typeof(FoamVisualizerSystem))]
+    public string State = "foam-dissolve";
+
+    [ViewVariables(VVAccess.ReadOnly)]
+    [Access(typeof(FoamVisualizerSystem))]
+    public Animation Animation = default!;
+}
+
+public sealed class FoamVisualizerSystem : VisualizerSystem<FoamVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<FoamVisualizerComponent, ComponentInit>(OnInit);
+    }
 
-        [DataField("animationState")]
-        private string _state = "foam-dissolve";
+    private void OnInit(EntityUid uid, FoamVisualizerComponent comp, ComponentInit args)
+    {
+        var flick = new AnimationTrackSpriteFlick();
+        flick.LayerKey = FoamVisualLayers.Base;
+        flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(comp.State, 0f));
+        comp.Animation = new Animation {Length = TimeSpan.FromSeconds(comp.Delay)};
+        comp.Animation.AnimationTracks.Add(flick);
```

Comment:
Make sure this and every animation are a single object initializer so they are easier to understand